tests/contrib/sanic/test_sanic.py

```diff
@@ -0,0 +1,170 @@
+import re
+
+import pytest
+from sanic import Sanic
+from sanic.exceptions import ServerError
+from sanic.response import json, stream
+
+import ddtrace
+from ddtrace.contrib.sanic import patch, unpatch
+from ddtrace.propagation import http as http_propagation
+from tests.base import BaseTestCase
+from tests.test_tracer import get_dummy_tracer
+
+
+@pytest.fixture
+def tracer():
+    original_tracer = ddtrace.tracer
+    tracer = get_dummy_tracer()
+    setattr(ddtrace, "tracer", tracer)
+    yield tracer
+    setattr(ddtrace, "tracer", original_tracer)
+
+
+@pytest.fixture
+def app(tracer):
+    patch()
+    app = Sanic(__name__)
+    yield app
+    unpatch()
+
+
+def test_basic_app(app, tracer):
+    """Test Sanic Patching"""
+
+    @app.route("/")
+    async def test(request):
+        return json({"hello": "world"})
+
+    request, response = app.test_client.get("/")
+    assert response.status == 200
+    assert response.body == b'{"hello":"world"}'
+
+    spans = tracer.writer.pop_traces()
+    assert len(spans) == 1
+    assert len(spans[0]) == 1
+    request_span = spans[0][0]
+    assert request_span.name == "sanic.request"
+    assert request_span.error == 0
+    assert request_span.get_tag("http.method") == "GET"
+    assert re.match("http://127.0.0.1:\\d+/", request_span.get_tag("http.url"))
```

Comment:
Any particular reason we're doing pattern matching versus a direct string comparison here? The URL should be static I think

Reply:
👋 @Kyle-Verhoog , by default, if port is not defined, it looks like [the test client](https://sanic.readthedocs.io/en/latest/sanic/testing.html?highlight#using-a-random-port) will choose the free random unprivileged port chosen by the kernel (On most systems the port will be in the range 1024 to 65535); therefore, we used pattern matching. Should we specify the exact port? 🙏 !

Reply:
ahh ok! :+1: nope that's fine!